pykern/pkio.py

```diff
@@ -18,11 +18,26 @@
 import re
 import shutil
 import six
+import tempfile
 
 #: used during unit testing see ``pykern.pkunit.save_chdir``
 pkunit_prefix = None
 
 
+def atomic_write(filename, content, mode='w'):
+    """Writes to file atomically
+
+    Args:
+        filename (str or py.path.Local): File to write to
+    """
+    with tempfile.NamedTemporaryFile(
+        mode=mode, dir=os.path.dirname(str(filename))) as f:
+        f.write(content)
+        if os.path.exists(str(filename)):
+            os.remove(str(filename))
```

Comment:
This operation defeats the purpose of the function. `rename` is atomic (even by py.path). This isn't atomic, because the remove happens then the link. 

Some people may say that remove and link are necessary, because the file may not be writable, but the directory might be writable. That's correct, but not the purpose of this particular call which is not to "write no matter what", but to "write atomically". This is in a safe environment from the perspective of "no matter what", but not safe from process exits or system crashes or other types of asynchronous events.

Reply:
Doh, of course.